File: xiaohashu-auth/src/main/java/com/tongji/xiaohashu/auth/constant/RedisKeyConstants.java
package com.tongji.xiaohashu.auth.constant;

/**
 * @author tongji
 * @time 2025/3/14 11:14
 * @description Redis Key 常量类
 */
public class RedisKeyConstants {
    /**
     * 验证码 KEY 前缀
     */
    private static final String VERIFICATION_CODE_KEY_PREFIX = "verification_code:";

    /**
     * 构建验证码 KEY
     */
    public static String buildVerificationCodeKey(String phone) {
        return VERIFICATION_CODE_KEY_PREFIX + phone;
    }

    /**
     * 小哈书全局 ID 生成器 KEY
     */
    public static final String XIAOHASHU_ID_GENERATOR_KEY = "xiaohashu.id.generator";

    /**
     * 用户角色数据 KEY 前缀
     */
    private static final String USER_ROLES_KEY_PREFIX = "user:roles:";

    /**
     * 构建验证码 KEY
     */
    public static String buildUserRoleKey(String phone) {
        return USER_ROLES_KEY_PREFIX + phone;
    }

    /**
     * 角色对应的权限集合 KEY 前缀
     */
    private static final String ROLE_PERMISSIONS_KEY_PREFIX = "role:permissions:";


    /**
     * 构建角色对应的权限集合 KEY
     */
    public static String buildRolePermissionsKey(Long roleId) {
        return ROLE_PERMISSIONS_KEY_PREFIX + roleId;
    }
}

File: xiaohashu-gateway/src/main/java/com/tongji/xiaohashu/gateway/constant/RedisKeyConstants.kt
package com.tongji.xiaohashu.gateway.constant

/**
 * @author tongji
 * @time 2025/3/16 14:15
 * @description Redis连接
 */
object RedisKeyConstants {
    /**
     * 用户角色数据 KEY 前缀
     */
    private const val USER_ROLES_KEY_PREFIX = "user:roles:"

    /**
     * 角色对应的权限集合 KEY 前缀
     */
    private const val ROLE_PERMISSIONS_KEY_PREFIX = "role:permissions:"

    /**
     * 构建角色对应的权限集合 KEY
     */
    @JvmStatic
    fun buildRolePermissionsKey(roleKey: String) = "$ROLE_PERMISSIONS_KEY_PREFIX$roleKey"

    /**
     * 构建用户-角色 KEY
     */
    @JvmStatic
    fun buildUserRoleKey(userId: Long) = "$USER_ROLES_KEY_PREFIX$userId"
}